unity/Holo/Assets/ModelClippingPlaneControl.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using HoloToolkit.Unity.InputModule;
+using HoloToolkit.Unity.UX;
+
+public class ModelClippingPlaneControl : MonoBehaviour, IClickHandler
+{
+    public GameObject ClippingPlane;
+    public GameObject ClippingPlaneModifyBtn;
+
+    public GameObject LoadedModel;
+
+    BoundingBoxRig clipPlaneQuadBbox;
+    Material currentModelMaterial;
+    Material currentModelMaterialClip;
+    
+    bool clipPlaneOn = false;
+    bool clipPlaneModifyActive = false;
+
+    void Start()
+    {
+        clipPlaneQuadBbox = ClippingPlane.GetComponent<BoundingBoxRig>();
+        ClippingPlane.GetComponent<HandDraggable>().enabled = false;
+
+        ResetState();
+    }
+
+    public void Click(GameObject clickObj)
+    {
+        Debug.Log("Clicked obj: " + clickObj.name);
+        switch (clickObj.name)
+        {
+            case "ButtonClipping":
+                ClickedClipPlane();
+                break;
+
+            case "ButtonClippingManipulator":
+                ClickedClipPlaneModify();
+                break;
+        }
+
+    }
+
+    void ClickedClipPlaneModify()
+    {
+        Debug.Log("Toggle modification of clip plane");
+        clipPlaneModifyActive = !clipPlaneModifyActive;
+        if (clipPlaneModifyActive)
+        {
+            clipPlaneQuadBbox.Activate();
+            ClippingPlane.GetComponent<HandDraggable>().enabled = true;
+        }
+        else
+        {
+            clipPlaneQuadBbox.Deactivate();
+            ClippingPlane.GetComponent<HandDraggable>().enabled = false;
+        }
+    }
+
+    void ClickedClipPlane()
+    { 
+        const string materialsPath = "Materials/";
+        GameObject currentModel = GameObject.Find("mainModel");
+        if (!currentModel)
+        {
+            Debug.Log("No model loaded for clipping plane");
+            clipPlaneOn = false;
+            return;
+        }
+

```

Comment:
Instead of finding the model by name, you should use `ModelWithPlate.cs` state. It has references to `InstanceParent` (GameObject in the scene), `instanceTransformation` (instantiated at runtime, child of `InstanceParent`), `instance` (instantiated at runtime by loading from asset bundle, child of `instanceTransformation`).

The clip plane should be attached to the proper one (not sure if it makes a difference).

In `ModelWithPlate.cs` the `instance` and `instanceTransformation` are read-only, but if necessary they could be exposed as read-only properties like `public GameObject Instance { get { return instance; } }`.